### PR TITLE
Upgraded to Jint v3

### LINKED ISSUE #84: Migration to Jint v3
Greetings,

Have there been any official efforts on migrating Jint v3? Or is that awaiting Jint v3 eventually leaving "beta" state?

I did some quick poking around and it looks like at minimum Jint has made a lot of member access modifiers more restrictive, including changing IConstructor from public to internal.

### PR DESCRIPTION
**Pull request by tomvanenckevort**
# Types of Changes

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## Description

As discussed in #84 this PR is to cover the upgrade of AngleSharp.Js to use Jint v3.

Changes made:
- Upgraded Nuget package reference of Jint to v3.0.0.
- Changed target framework of project to .NET Standard 2.0 and .NET Framework 4.6.2 to match the Jint v3 targets.
- Updated Jint namespaces and class names (e.g. `FunctionInstance` > `Function`) to match the changes made in Jint v3.
- Updated `EngineInstance` to remove the lexical and variable environments since those are no longer public in Jint v3. Instead there is now a global `window` object which also gets exposed on the global object (to ensure that things like `window.location.href` and `location.href` resolve to the same thing).
- Replaced `ConsoleInstance` with a new `Console` class that ties it directly to the Window object (as per [JS spec](https://developer.mozilla.org/en-US/docs/Web/API/console)).
- Removed obsolete classes that have been replaced by Jint equivalents (e.g. `DomDelegateInstance` > `ClrFunction`).
- Checked that all existing unit tests are passing, and added a new ECMA test class. I've only added a single test in there to check that loading an ECMA script succeeds, but feel free to suggest any other tests required.

Also a shoutout to @lahma for providing some guidance and older test code that helped me figure out the global object changes. 👍

### CODE REVIEW

File: src/AngleSharp.Js/AngleSharp.Js.csproj [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
   <ItemGroup>
     <PackageReference Include="AngleSharp" Version="0.15.0" />
```
Comment (FlorianRappl) [inline src/AngleSharp.Js/AngleSharp.Js.csproj]: The big question is - should we make this AngleSharp 1.0.0 compatible?

We can also first release it for an older version and then upgrade it (making it also a 1.0.0).

**Review by FlorianRappl — COMMENTED** [review]
This is superb - you are officially a rockstar 🚀 !

What are you plans with this? How should we proceed to bring this live?

**Comment by FlorianRappl** [comment]
Also another question - what about #87 ? Should we bring this in (the test and fix, if that still applies)?

File: src/AngleSharp.Js/AngleSharp.Js.csproj (hunk 1 above)
Comment (tomvanenckevort) [reply src/AngleSharp.Js/AngleSharp.Js.csproj]: I think it makes sense to target AngleSharp 1.0.0 and then make this a v1.0.0 as well.
I'll make those changes and update my PR.

**Comment by tomvanenckevort** [comment]
I think #87 should be merged in `devel` first since it's been around for a while, and then I can rebase my branch on that one to ensure my changes work with that fix.

File: src/AngleSharp.Js/AngleSharp.Js.csproj (hunk 1 above)
Comment (tomvanenckevort) [reply src/AngleSharp.Js/AngleSharp.Js.csproj]: What about target frameworks? I notice that for AngleSharp v1 you target `netstandard2.0;net461;net472;net6.0;net7.0;net8.0`.
Should we aim for the same on this project?

Comment (FlorianRappl) [reply src/AngleSharp.Js/AngleSharp.Js.csproj]: Aim yes, but generally libs don't need to have the same scope, e.g., `AngleSharp.Io` has a few things that require "more" capabilities - so it is not as general.

Appreciated that you take the update!

**Comment by FlorianRappl** [comment]
Merged #87 into `devel`. Thanks for taking this and putting in the effort @tomvanenckevort !

**Review by FlorianRappl — APPROVED** [review]
Looks awesome to me. Let's have the build green and have the first pre-release done.

**Comment by tomvanenckevort** [comment]
We're note quite there yet.
I ran some extra tests locally yesterday and realised that the `<script type="module">` tags currently don't work. So I need to have another look at that one.
But I guess we can handle that via another PR :)

**Comment by FlorianRappl** [comment]
Yes, I mean it's anyway just a first preview. I've assigned some / relevant issues to the v1 milestone.

I am not sure yet how much I can look into them tonight and on the weekend, but my goal would be to have a bit of focus now on AngleSharp.Js to have this as a 1.0.0. Thanks for your PR and motivating me to do so.

If you want to pick up / anyway have some of these issues handled then please comment on them. Otherwise I'll just pick up whatever is left.

**Comment by FlorianRappl** [comment]
Preview is out (https://www.nuget.org/packages/AngleSharp.Js/0.15.0-beta.16).

The version is still placed at 0.15 and I also noticed that some parts of the NuGet package definition (e.g., the README) are missing: I'll add those and push to `devel`.

**Comment by lahma** [comment]
Awesome work @tomvanenckevort ! I'll hopefully have some timesoon  too to check this out and investigate if there's something that Jint could improve too. Just have to get back from vacation first 😄

**Comment by FlorianRappl** [comment]
Preview is out (https://www.nuget.org/packages/AngleSharp.Js/1.0.0-beta.20); incl. latest target frameworks, correct reference to Jint and AngleSharp, README etc.

I think from the infrastructure POV we are done, now we "only" need to clear the outstanding issues and have it (*draw the rest of the owl*).

How to draw an owl

**Pull request merged**